static/app/utils/duration/formatDuration.tsx

```diff
@@ -82,6 +86,49 @@ export default function formatDuration({
       });
       const [head, tail] = str.split('.');
       return includeMs ? [head, tail ?? '000'].join('.') : String(head);
+    case 'ISO8601':
+      const output = ['P'];
+
+      let incr = 0;
+      const weeks = Math.floor(msToPrecision(ms - incr, 'week'));
+      output.push(weeks ? weeks + 'W' : '');
+      if (precision !== 'week') {
+        incr += weeks * PRECISION_FACTORS.week;
```

Comment:
wonder how much precision matters here since it isn't really formatted for humans

Reply:
Ya, it's mostly for completeness, and the logic is the same as the magnitude goes up so not really any harm imo.

I also think the interface of asking for precision is scalable. For context our existing `getDuration()` and `getExactDuration()` functions are next to merge into here. I think they would both transform into a new format `format="human-full"` and the difference between exact and non-exact will be a matter of asking for the right precision (a helper util will have a place here).
Also, to account for the `abbreviation=true` param another `format="human-abbr"` (might not be the final names, but that idea) can be put in.